### DISCUSSION TITLE
Running parallel tests

### DISCUSSION THREAD

**Discussion by Vladelis** [spatie/laravel-event-sourcing#392] [Q&A] [answered]
Trying to make use of the new [Laravel Parallel Testing](https://laravel.com/docs/9.x/testing#running-tests-in-parallel) work, but the run fails with the decline reason

```
⇒  php artisan test --parallel

In Container.php line 1118:

  Unresolvable dependency resolving [Parameter #0 [ <required> string $storedEventRepository ]] in class Spatie\EventSourcing\StoredEven
  ts\EventSubscriber

``` 

Looks like `EventSubscribers` could not be initialised properly for parallel testing.

Has anyone come up with a solution to make this work? Perhaps we could see this being solved in a foreseeable future?

Parallel test work upon removing event sourcing from the project.

Running these versions and dependencies:

`phpunit/phpunit 9.6.3`
`pestphp/pest 1.22`
`pestphp/pest-plugin-parallel 1.2`
`spatie/laravel-event-sourcing 7.3.3` 
`laravel/framework 9.5` 

Thank you.

**Comment by Sparclex** [comment]
I am running into the same problem now. Did you find a solution for it? @Vladelis

**Reply by Vladelis** [reply]
We ended up deprecating event sourcing on the project where we had this issue. Strangely enough we **did not** have the same issue on other projects using ES as parallel tests seemed to run fine.

**Reply by Sparclex** [reply, marked as answer]
I found my mistake. I used ﻿pest and initialized a dummy class, which inadvertently initialized some event sourcing classes, outside of all functions. As a result, the class could not be instantiated since the container was not defined.

**Reply by sebastiandedeyne** [reply]
Sounds like a viable answer. In the future, I hope to reduce framework dependencies for some parts so these issues become less common.
